manifests/config.pp

```diff
@@ -21,13 +21,24 @@
     "main_collective" => $main_collective
   }
 
+  # some config settings are set directly on the main class, they are
+  # things other classes need to function for example libdir is also
+  # used by the plugin installer.  Specifically these settings should
+  # never be overridden by settings from the hashes since any difference
+  # between server/client cfg and what ends up being used by the plugin
+  # installer will lead to plugins not being found.
+  $global_config = {
```

Comment:
yeah :( more piles of shit, this should fix the immediate problem we can always refactor later.

As it is if the libdirs fall out of sync nothing will work, so thats the main issue I am addressing here, later more can be added